### PR TITLE
CLDC-2938 Add sales managing organisation question

### PR DESCRIPTION
**Pull request by kosiakkatrina**
We want to start collecting managing (reporting) organisation for sales logs as well.
We would probably want to add the column and update the managing organisation for records to be the same as the owning organisation first, to avoid any data loss due to routing, so this should be merged and ran first separately: https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/2048

In this PR:
- Add managing organisation question which we only display to support users, otherwise it gets inferred as current users organisation. 
- Route to owning organisation question if the org has more than 1 stock owner (including itself).
- Created by has to be someone from the owning organisation.
- Add reported by and owned by filters where appropriate
- Add reported by and owned by tags to the logs list view

### CODE REVIEW

File: app/controllers/lettings_logs_controller.rb [hunk 1]
```diff
@@ -123,12 +123,6 @@ def filter_manager
     FilterManager.new(current_user:, session:, params:, filter_type: "lettings_logs")
   end
 
-  def org_params
```
Comment (natdeanlewissoftwire) [inline app/controllers/lettings_logs_controller.rb:126]: nice!

File: spec/factories/sales_log.rb [hunk 2]
```diff
@@ -2,6 +2,7 @@
   factory :sales_log do
     created_by { FactoryBot.create(:user) }
     owning_organisation { created_by.organisation }
+    managing_organisation { owning_organisation }
```
Comment (natdeanlewissoftwire) [inline spec/factories/sales_log.rb:5]: would have assumed this doesn't work, but nice :)

File: app/models/form/sales/pages/organisation.rb [hunk 3]
```diff
@@ -10,24 +10,40 @@ def questions
...
-    if FeatureToggle.merge_organisations_enabled?
+  def routed_to?(log, current_user)
+    return false unless current_user
+    return true if current_user.support?
+    return false unless FeatureToggle.sales_managing_organisation_enabled?
 
-      return false unless current_user
-      return true if current_user.support?
+    return true if stock_owners_with_own_stock_count(current_user) > 1
+    return true if current_user.organisation.holds_own_stock? && stock_owners_with_own_stock_count(current_user) >= 1
```
Comment (natdeanlewissoftwire) [inline app/models/form/sales/pages/organisation.rb]: can we combine this and the previous line with a new method like `manages_logs_for_multiple_stock_owners?(current_user)` for clarity?

File: app/models/form/sales/pages/organisation.rb [hunk 4]
```diff
@@ -10,24 +10,40 @@ def questions
...
+
+      log.update!(owning_organisation: stock_owners.first)
     end
+
+    false
+  end
+
+private
+
+  def stock_owners_with_own_stock_count(user)
```
Comment (natdeanlewissoftwire) [inline app/models/form/sales/pages/organisation.rb]: Can we move this out of the page to a shared location for sales and lettings? Think I might have been the one to put the equivalent method on the lettings page, sorry if that's true!

File: app/models/form/sales/pages/organisation.rb
Comment (natdeanlewissoftwire) [inline app/models/form/sales/pages/organisation.rb:1]: think we should rename this file to be `owning_organisation`

Comment (natdeanlewissoftwire) [reply app/models/form/sales/pages/organisation.rb:1]: and the id etc etc.

**Review by natdeanlewissoftwire — COMMENTED** [review]
Looks good, mostly refactoring thoughts that I've commented - one larger thought is:

We've got a lot of validations that reference managing_organisation in the lettings log setup/financial validations files. Do we want to add similar validations for any/all of these to sales?

Also, there are some validations in bulk upload that check e.g. user is in either the owning or managing org, I think we should be adding these to the sales row parser too now.

File: app/models/form/sales/pages/organisation.rb (hunk 3 above)
Comment (kosiakkatrina) [reply app/models/form/sales/pages/organisation.rb]: Sure!

File: app/models/form/sales/pages/organisation.rb (hunk 4 above)
Comment (kosiakkatrina) [reply app/models/form/sales/pages/organisation.rb]: Ended up removing this altogether

File: app/models/form/sales/pages/organisation.rb
Comment (kosiakkatrina) [reply app/models/form/sales/pages/organisation.rb:1]: Done!

**Comment by kosiakkatrina** [comment]
> Looks good, mostly refactoring thoughts that I've commented - one larger thought is:
> 
> We've got a lot of validations that reference managing_organisation in the lettings log setup/financial validations files. Do we want to add similar validations for any/all of these to sales?
> 
> Also, there are some validations in bulk upload that check e.g. user is in either the owning or managing org, I think we should be adding these to the sales row parser too now.

I think financial validations are around rent periods so probably not relevant to sales. The setup validations were around the merged orgs stuff which should be there and I've now updated the bulk upload logic! Thanks!

**Review by natdeanlewissoftwire — APPROVED** [review]

**Pull request closed without merging**